go/cmd/dolt/commands/sql.go

```diff
@@ -202,6 +204,47 @@ func (cmd SqlCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 	return 0
 }
 
+func formatProcessQueryError(err error) errhand.VerboseError {
+	const (
+		maxStatementLen = 128
+		maxPosWhenTruncated = 64
+	)
+
+	verrBuilder := errhand.BuildDError("Error parsing SQL")
```

Comment:
This isn't always accurate -- the methods that return syntax errors also return other types of errors. You need to move this inside the type check